x/mongo/driver/topology/server_test.go

```diff
@@ -309,6 +312,90 @@ func TestServer(t *testing.T) {
 			t.Fatal("client metadata not expected in heartbeat but found")
 		}
 	})
+	t.Run("heartbeat monitoring", func(t *testing.T) {
+		var publishedEvents []interface{}
+
+		serverHeartbeatStarted := func(e *event.ServerHeartbeatStartedEvent) {
+			publishedEvents = append(publishedEvents, *e)
+		}
+
+		serverHeartbeatSucceeded := func(e *event.ServerHeartbeatSucceededEvent) {
+			publishedEvents = append(publishedEvents, *e)
+		}
+
+		serverHeartbeatFailed := func(e *event.ServerHeartbeatFailedEvent) {
+			publishedEvents = append(publishedEvents, *e)
+		}
+
+		sdam := &event.ServerMonitor{
+			ServerHeartbeatStarted:   serverHeartbeatStarted,
+			ServerHeartbeatSucceeded: serverHeartbeatSucceeded,
+			ServerHeartbeatFailed:    serverHeartbeatFailed,
+		}
+
+		// test that client metadata is sent on handshakes but not heartbeats
```

Comment:
I think this comment is copy/pasted from another test?

Reply:
done